processor/src/chiplets/hasher/tests.rs

```diff
@@ -534,14 +968,61 @@ fn hash_memoization_span_blocks_check(span_block: CodeBlock) {
         unreachable!()
     };
 
+    let mut lookups = Vec::new();
     // builds the hash execution trace of the second span block by copying the sections of the
     // trace corresponding to the first span block with the same hash.
-    let (addr, final_state, _) = hasher.hash_span_block(
+    let (addr, final_state) = hasher.hash_span_block(
         span2_block_val.op_batches(),
         get_span_op_group_count(span2_block_val.op_batches()),
         span2_block.hash(),
+        &mut lookups,
     );
 
+    let num_batches = if let CodeBlock::Span(span) = &span2_block {
+        span.op_batches().len()
+    } else {
+        unreachable!()
+    };
+
+    let expected_lookups_len = 2 + num_batches - 1;
+    // make sure the lookups have correct labels, addresses, indices and contexts.
+    let expected_lookups = if num_batches == 1 {
+        vec![
+            HasherLookup::new(LINEAR_HASH_LABEL, 25, ZERO, HasherLookupContext::Start),
+            HasherLookup::new(
+                RETURN_HASH_LABEL,
+                25 + HASH_CYCLE_LEN as u32 - 1,
+                ZERO,
+                HasherLookupContext::Return,
+            ),
+        ]
+    } else {
+        // the start row is 41 here instead of 25 since the hash execution trace of the last
+        // (memoized) span block ends at row 40 in case of span with multiple batches.
```

Comment:
nit: I'd prefer using named variables for first hash and span block length and then computing it. It makes things less opaque and more resilient if these tests change, and makes this comment less necessary

Reply:
Fixed.